cpp/src/spatial/hausdorff.cu

```diff
@@ -103,49 +135,92 @@ struct hausdorff_functor {
   std::enable_if_t<std::is_floating_point<T>::value, std::unique_ptr<cudf::column>> operator()(
     cudf::column_view const& xs,
     cudf::column_view const& ys,
-    cudf::column_view const& points_per_space,
+    cudf::column_view const& space_offsets,
     rmm::mr::device_memory_resource* mr,
     cudaStream_t stream)
   {
-    auto tid    = cudf::type_to_id<T>();
-    auto result = cudf::make_fixed_width_column(cudf::data_type{tid},
-                                                points_per_space.size() * points_per_space.size(),
-                                                cudf::mask_state::UNALLOCATED,
-                                                stream,
-                                                mr);
+    size_type num_points  = xs.size();
+    size_type num_spaces  = space_offsets.size();
+    size_type num_results = num_spaces * num_spaces;
+
+    if (num_results == 0) { return make_column<T>(0, stream, mr); }
 
-    if (result->size() == 0) { return result; }
+    // ===== Make Lookup for Space by Point ========================================================
+    // these space lookups could be replaced with a `lower_bound` to reduce temporary memory
 
-    auto space_offsets = rmm::device_vector<cudf::size_type>(points_per_space.size());
+    auto temp_space_lookup = rmm::device_vector<size_type>(num_points);
+
+    thrust::scatter(rmm::exec_policy(stream)->on(stream),
+                    thrust::make_constant_iterator(1),
+                    thrust::make_constant_iterator(1) + num_spaces - 1,
+                    space_offsets.begin<size_type>() + 1,
+                    temp_space_lookup.begin());
 
     thrust::inclusive_scan(rmm::exec_policy(stream)->on(stream),
-                           points_per_space.begin<cudf::size_type>(),
-                           points_per_space.end<cudf::size_type>(),
-                           space_offsets.begin());
+                           temp_space_lookup.cbegin(),
+                           temp_space_lookup.cend(),
+                           temp_space_lookup.begin());
+
+    // ===== Make Space Size Iterator ==============================================================
+
+    auto d_space_offsets = cudf::column_device_view::create(space_offsets);
+
+    auto space_offset_iterator =
+      thrust::make_transform_iterator(thrust::make_counting_iterator<int32_t>(0),
+                                      size_from_offsets_functor{*d_space_offsets, xs.size()});
+
+    // ===== Make Hausdorff Accumulator ============================================================
 
-    // utilize one block per result (pair of spaces).
-    size_type num_blocks_x = min(result->size(), MAX_NUM_BLOCKS_X);
-    size_type num_blocks_y = ceil(result->size() / (float)MAX_NUM_BLOCKS_X);
+    auto num_distances = static_cast<int32_t>(num_points) * static_cast<int32_t>(num_points);
```

Comment:
Why do you need these casts?

Reply:
Don't need them. Leftover from testing. Thanks for catching it. :smiley: